gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -114,10 +115,47 @@ class KspKotlinGradleSubplugin : KotlinGradleSubplugin<AbstractCompile> {
             javaCompile.source(generatedJavaSources)
         }
 
-        return options
+        val kspTaskName = kotlinCompile.name.replaceFirst("compile", "ksp")
+        InternalTrampoline.KotlinCompileTaskData_register(kspTaskName, kotlinCompilation)
+
+        val kspTaskProvider = project.tasks.register(kspTaskName, KspTask::class.java) { kspTask ->
+            kspTask.setDestinationDir(File(project.buildDir, "generated/ksp"))
+            kspTask.mapClasspath { kotlinCompile.classpath }
+            kspTask.options = options
+        }.apply {
+            configure {
+                kotlinCompilation?.allKotlinSourceSets?.forEach { sourceSet -> it.source(sourceSet.kotlin) }
+            }
+        }
+
+        kotlinCompile.dependsOn(kspTaskProvider)
+        kotlinCompile.source(kotlinOutputDir, javaOutputDir)
+
+        return emptyList()
     }
 
-    override fun getCompilerPluginId() = "com.google.devtools.ksp.symbol-processing"
+    override fun getCompilerPluginId() = KSP_PLUGIN_ID
     override fun getPluginArtifact(): SubpluginArtifact =
         SubpluginArtifact(groupId = "com.google.devtools.ksp", artifactId = KSP_ARTIFACT_NAME, version = javaClass.`package`.implementationVersion)
 }
+
+open class KspTask : KotlinCompile() {
+    lateinit var options: List<SubpluginOption>
+
+    init {
+        // kotlinc's incremental compilation isn't compatible with symbol processing in a few ways:
+        // * It doesn't consider private / internal changes when computing dirty sets.
+        // * It compiles iteratively; Sources can be compiled in different rounds.
+        incremental = false
```

Comment:
The PR description says it was reenabled but this seems to suggest it's not?

Reply:
Incremental compilation is back on in the compilation Gradle task, but incremental processing is off in the new ksp specific Gradle task, as the later is still under construction :-)